### PR TITLE
feat: Enable unprivileged ports sysctl in containerd config

### PR DESCRIPTION
**Pull request by jimmidyson**
This enabled pods to run as non-root and bind to privileged ports
as long as they have the necessary capability, `CAP_NET_BIND_SERVICE`
added.

This fixes an issue on AWS when bringing up coredns which binds to port
53 but runs as an unprivileged user.

Overall this is a net security improvement for clusters, meaning users
can stop giving too many privileged to pods - see
https://github.com/kubernetes/kubernetes/issues/102612 for discussion.

Depends on #644.

### CODE REVIEW

**Comment by dlipovetsky** [comment]
CoreDNS v1.11.1 is deployed with Kubernetes v1.29.0.

Its securityContext stays the same as before:

```yaml
    securityContext:
      allowPrivilegeEscalation: false
      capabilities:
        add:
        - NET_BIND_SERVICE
        drop:
        - all
      readOnlyRootFilesystem: true
```

But it now run as a nonroot user (https://github.com/coredns/coredns/pull/5969).

However, even with this capability, containerd does not allow binding to an unprivileged port. For that, containerd must be configured as this PR does it.

For comparison, see an identical change to containerd configuration in https://github.com/kubernetes/minikube/pull/18421

**Review by dlipovetsky — APPROVED** [review]

**Review by faiq — DISMISSED** [review]

**Review by faiq — APPROVED** [review]

File: test/e2e/config/caren.yaml [hunk 1]
```diff
@@ -184,15 +184,15 @@ variables:
   AMI_LOOKUP_ORG: "999867407951"
   # To run Nutanix provider tests, set following variables here or as an env var
   # # IP/FQDN of Prism Central.
-  # NUTANIX_ENDPOINT: ""
+  NUTANIX_ENDPOINT: ""
```
Comment (dkoshkin) [inline test/e2e/config/caren.yaml:187]: Was this change intentional?

Comment (jimmidyson) [reply test/e2e/config/caren.yaml:187]: Not for this PR 😅 will revert

Comment (jimmidyson) [reply test/e2e/config/caren.yaml]: Fixed.

File: pkg/handlers/generic/mutation/containerdunprivilegedports/unprivileged_ports.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+import (
+	_ "embed"
+	"path"
+
+	cabpkv1 "sigs.k8s.io/cluster-api/bootstrap/kubeadm/api/v1beta1"
+)
+
+const (
+	// TODO Factor out this constant to a common package.
```
Comment (dkoshkin) [inline pkg/handlers/generic/mutation/containerdunprivilegedports/unprivileged_ports.go:13]: Is this PR enough https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/656/files?

**Review by dkoshkin — APPROVED** [review]
Thanks for figuring this out!

File: pkg/handlers/generic/mutation/containerdunprivilegedports/unprivileged_ports.go (hunk 2 above)
Comment (jimmidyson) [reply pkg/handlers/generic/mutation/containerdunprivilegedports/unprivileged_ports.go]: Good catch! Thanks! Updated to use new common function to derive patch file path.
